target/linux/ipq807x/base-files/lib/upgrade/platform.sh

```diff
@@ -66,6 +66,18 @@ platform_do_upgrade() {
 		rootfsname="rootfs"
 		mmc_do_upgrade "$1"
 		;;
+	zyxel,nbg7815)
+		part_num="$(hexdump -e '1/1 "%01x|"' -n 1 -s 168 -C /dev/mtd2|cut -f 1 -d "|"|head -n1)"
```

Comment:
Please no partition hardcoding, there are generic helpers to find the mtd index by partition name instead